chain/client/src/client.rs

```diff
@@ -1053,9 +1056,9 @@ impl Client {
         let mut partial_encoded_chunks =
             self.shards_mgr.get_stored_partial_encoded_chunks(next_height);
         for (_shard_id, partial_encoded_chunk) in partial_encoded_chunks.drain() {
-            if let Ok(accepted_blocks) =
-                self.process_partial_encoded_chunk(PartialEncodedChunk::V2(partial_encoded_chunk))
-            {
+            let chunk =
+                MaybeValidated::NotValidated(PartialEncodedChunk::V2(partial_encoded_chunk));
```

Comment:
why is this not validated?

Reply:
My understanding is that this covers the case where we were missing some chunks for the block at height `h`, so could not accept it. But in the mean time we have received the block and some chunks for height `h+1`, which will need to be processed once we finish with height `h`. And it looked to me like the signature check does not happen for the `h+1` because they point at a block we haven't added yet. Therefore these should be labelled as `NotValidated`.

Reply:
I see